ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTable.java

```diff
@@ -76,15 +92,30 @@
   @Override
   public Iterator<Row> get(final int partition, final GenericKey from, final GenericKey to) {
     try {
-      final ReadOnlyKeyValueStore<GenericKey, ValueAndTimestamp<GenericRow>> store = stateStore
-          .store(QueryableStoreTypes.timestampedKeyValueStore(), partition);
+      final RangeQuery<GenericKey, ValueAndTimestamp<GenericRow>> query;
+      if (from != null && to != null) {
+        query = RangeQuery.withRange(from, to);
+      } else if (from == null && to != null) {
+        query = RangeQuery.withUpperBound(to);
+      } else if (from != null && to == null) {
+        query = RangeQuery.withLowerBound(from);
+      } else {
+        query = RangeQuery.withNoBounds();
+      }
 
+      final StateQueryRequest<KeyValueIterator<GenericKey, ValueAndTimestamp<GenericRow>>>
+          request = inStore(stateStore.getStateStoreName())
+          .withQuery(query)
+          .withPartitions(ImmutableSet.of(partition));
+      final StateQueryResult<KeyValueIterator<GenericKey, ValueAndTimestamp<GenericRow>>>
+          result = stateStore.getKafkaStreams().query(request);
       final KeyValueIterator<GenericKey, ValueAndTimestamp<GenericRow>> iterator =
-          store.range(from, to);
+          result.getOnlyPartitionResult().getResult();
```

Comment:
We don't really need to do "get only partition result" because we already know the partition we requested. I doesn't matter much, but it seems more direct to say:

```suggestion
          result.getPartitionResults().get(partition).getResult();
```

(Also applies to the other query)

Reply:
Already on it :) Unfortunately, you reviewed a stale version